anise/src/naif/daf/mod.rs

```diff
@@ -45,6 +45,7 @@ pub trait NAIFRecord: AsBytes + FromBytes + Sized + Default + Debug {
 
 pub trait NAIFSummaryRecord: NAIFRecord + Copy {
     fn start_index(&self) -> usize;
+    fn data_type_i(&self) -> i32;
```

Comment:
Instead of adding the `data_type_i`, I recommend making the `data_type()` function of [SPKSummaryRecord](https://docs.rs/anise/latest/anise/naif/spk/summary/struct.SPKSummaryRecord.html#method.data_type) and [BPCSummaryRecord](https://docs.rs/anise/latest/src/anise/naif/pck/mod.rs.html#39-44) generic.

The way you could do this is by adding a new type parameter to the trait as follows:

```rust
us core::error::Error;

pub trait NAIFSummaryRecord: NAIFRecord + Copy {
    type ErrorType: Error;
    /// (...)
    fn data_type(&self) -> Result<DafDataType, Self::ErrorType>;
}
```

Then in the BPCSummaryType, move the data_type function from this struct's own implementation to the struct's implementation of the NAIFSummaryRecord trait.

So around line 160 of naif/spk/summary.rs, you'd need to add the `type Error` definition.

```rust
impl NAIFSummaryRecord for SPKSummaryRecord {
    const NAME: &'static str = "SPKSummaryRecord";
    type ErrorType = EphemerisError;
    // (...)
   fn data_type(&self) -> Result<DafDataType, EphemerisError> {
        DafDataType::try_from(self.data_type_i).map_err(|source| EphemerisError::SPK {
            action: "converting data type from i32",
            source,
        })
    }
```

Repeat that for the BPC Summary structure.

Then in the CLI code, you can just use the data_type() function of the NAIFSummary directly, and remove the need to export the integer representing the data_type_i or even rebuilding the data type enum from that integer in the code.

Let me know if you'd like more clarifications here.